src/server/hotkey_collector.cpp

```diff
@@ -49,5 +67,51 @@ void hotkey_collector::capture_hash_key(const dsn::blob &hash_key, int64_t weigh
 
 void hotkey_collector::analyse_data() { _internal_collector->analyse_data(); }
 
+void hotkey_collector::on_start_detect(dsn::replication::detect_hotkey_response &resp)
+{
+    auto now_state = _state.load();
+    std::string hint;
+    switch (now_state) {
+    case hotkey_collector_state::COARSE_DETECTING:
+    case hotkey_collector_state::FINE_DETECTING:
+        resp.err = dsn::ERR_SERVICE_ALREADY_EXIST;
+        hint = fmt::format("still detecting {} hotkey, state is {}",
+                           dsn::enum_to_string(_hotkey_type),
+                           enum_to_string(now_state));
+        ddebug_replica(hint);
```

Comment:
I think warning or error level may be better, the following log-printing also has this problem.